### PR TITLE
FB App Events: version migration to v13.0

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> version migration to v13.0
https://developers.facebook.com/docs/graph-api/changelog/versions

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/fb/transform.js [hunk 1]
```diff
@@ -253,8 +248,11 @@ function buildBaseEvent(message) {
   } else if (isAppleFamily(sourceSDK)) {
     sourceSDK = "i2";
   } else {
-    // if the sourceSDK is not android or ios, send an empty string
-    sourceSDK = "";
+    // if the sourceSDK is not android or ios
+    throw new CustomError(
+      "Extended Device Information i.e device.type is required",
+      400
+    );
```
Comment (ItsSudip) [inline v0/destinations/fb/transform.js:255]: This is a breaking change we should provide support for backward compatibility.

Comment (utsabc) [reply v0/destinations/fb/transform.js:255]: The error from facebook is not quite understandable. Hence we are blocking at transformer with understandable error message. Either way this is a breaking change introduced by facebook

**Review by utsabc — DISMISSED** [review]

**Review by ItsSudip — DISMISSED** [review]

File: v0/destinations/fb/transform.js [hunk 2]
```diff
@@ -253,8 +248,11 @@ function buildBaseEvent(message) {
   } else if (isAppleFamily(sourceSDK)) {
     sourceSDK = "i2";
   } else {
-    // if the sourceSDK is not android or ios, send an empty string
-    sourceSDK = "";
+    // if the sourceSDK is not android or ios
+    throw new CustomError(
+      "Extended Device Information i.e device.type is required",
```
Comment (saikumarrs) [inline v0/destinations/fb/transform.js:253]:
```suggestion
      "Extended device information i.e, "context.device.type" is required",
```

File: v0/destinations/fb/transform.js [hunk 3]
```diff
@@ -220,16 +220,11 @@ function processEventTypeGeneric(message, baseEvent, fbEventName) {
...
 
 function responseBuilderSimple(message, payload, destination) {
-  const requestConfig = {
-    requestFormat: "FORM",
-    requestMethod: "POST"
-  };
-
   const { appID } = destination.Config;
 
   // "https://graph.facebook.com/v3.3/644758479345539/activities?access_token=644758479345539|748924e2713a7f04e0e72c37e336c2bd"
```
Comment (saikumarrs) [inline v0/destinations/fb/transform.js:225]: Let's update this comment too

Comment (saikumarrs) [reply v0/destinations/fb/transform.js:225]: Also, make sure that the data in it is invalid.

File: v0/destinations/fb/transform.js [hunk 4]
```diff
@@ -253,8 +248,11 @@ function buildBaseEvent(message) {
...
-    // if the sourceSDK is not android or ios, send an empty string
-    sourceSDK = "";
+    // if the sourceSDK is not android or ios
+    throw new CustomError(
+      "Extended Device Information i.e device.type is required",
+      400
+    );
   }
 
   baseEvent.extinfo[0] = sourceSDK;
```
Comment (saikumarrs) [inline v0/destinations/fb/transform.js:258]: Remove the commented code from Ln 295.

Comment (ujjwal-ab) [reply v0/destinations/fb/transform.js:258]: Done

File: v0/destinations/fb/transform.js (hunk 3 above)
Comment (ujjwal-ab) [reply v0/destinations/fb/transform.js]: Updated

File: v0/destinations/fb/transform.js (hunk 2 above)
Comment (ujjwal-ab) [reply v0/destinations/fb/transform.js]: Done

File: v0/destinations/fb/transform.js [hunk 5]
```diff
@@ -253,8 +248,11 @@ function buildBaseEvent(message) {
   } else if (isAppleFamily(sourceSDK)) {
     sourceSDK = "i2";
   } else {
-    // if the sourceSDK is not android or ios, send an empty string
-    sourceSDK = "";
+    // if the sourceSDK is not android or ios
+    throw new CustomError(
+      "Extended Device Information i.e context.device.type is required",
```
Comment (saikumarrs) [inline v0/destinations/fb/transform.js:253]:
```suggestion
      "Extended device information i.e, "context.device.type" is required",
```

Comment (ujjwal-ab) [reply v0/destinations/fb/transform.js]: Done

File: v0/destinations/fb/transform.js [hunk 6]
```diff
@@ -253,8 +248,11 @@ function buildBaseEvent(message) {
   } else if (isAppleFamily(sourceSDK)) {
     sourceSDK = "i2";
   } else {
-    // if the sourceSDK is not android or ios, send an empty string
-    sourceSDK = "";
+    // if the sourceSDK is not android or ios
+    throw new CustomError(
+      'Extended Device Information i.e, "context.device.type" is required',
```
Comment (saikumarrs) [inline v0/destinations/fb/transform.js:253]: Just directly commit this suggestion,
```suggestion
      'Extended device information i.e, "context.device.type" is required',
```

Comment (ujjwal-ab) [reply v0/destinations/fb/transform.js]: Done

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**